includes/class-amp-theme-support.php

```diff
@@ -1497,6 +1497,38 @@ protected static function has_dependency( WP_Dependencies $dependencies, $curren
 		return false;
 	}
 
+	/**
+	 * Check if a handle is exclusively a dependency of another handle.
+	 *
+	 * For example, check if dashicons is being added exclusively because it is a dependency of admin-bar, as opposed
+	 * to being added because it was directly enqueued by a theme or a dependency of some other style.
+	 *
+	 * @since 1.4.2
+	 *
+	 * @param WP_Dependencies $dependencies         Dependencies.
+	 * @param string          $handle               Handle.
+	 * @param string          $exclusive_dependency Exclusive dependency handle.
+	 * @return bool Whether the $handle is exclusively a handle of the $exclusive_dependency handle.
+	 */
+	protected static function is_exclusive_dependency( WP_Dependencies $dependencies, $handle, $exclusive_dependency ) {
+		foreach ( $dependencies->queue as $queued_handle ) {
+			if (
+				// If a theme or plugin directly enqueued handle, then it is not added via $exclusive_dependency and it is not part of dev mode.
```

Comment:
The comments here should not refer to the dev-mode, as the code that was now extracted is not directly concerned with dev-mode anymore.